### PR TITLE
added slack notification details

### PR DESCRIPTION
**Pull request by lingpri**
fixes PR #625 

slack_notification_details

### CODE REVIEW

File: src/web/index.jsx
Comment (waab76) [inline src/web/index.jsx:15]: I expect the linter to complain about this line.  Should probably be between `PagerDutyNotificationSummary` and `SlackNotificationForm`

File: src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx
Comment (waab76) [inline src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx:19]: Is there a reason the other options are not being displayed?

File: src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx
Comment (waab76) [inline src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx:23]: I believe our front end engineers would be happier with a fully-specified shape here:
```suggestion
  notification: PropTypes.shape({
    config: PropTypes.shape({
      graylog_url: PropTypes.string,
      icon_emoji: PropTypes.string,
      icon_url: PropTypes.string,
      link_names: PropTypes.string,
      notify_channel: PropTypes.string,
      user_name: PropTypes.string,
      custom_message: PropTypes.string,
      channel: PropTypes.string,
      webhook_url: PropTypes.string,
      color: PropTypes.string,
    }).isRequired,
```

**Review by waab76 — CHANGES_REQUESTED** [review]
Please also get a review from someone with frontend knowledge (@kyleknighted @dennisoelkers @edmundoa @Gaya)

File: src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx
Comment (kyleknighted) [inline src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx:6]: should be a styled-component

File: src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx
Comment (kyleknighted) [inline src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx:5]: Something is up with the eslint config on this repo because there are lots of lint errors I can spot

File: src/web/event-notifications/event-notification-details/SlackNotificationSummary.css
Comment (kyleknighted) [inline src/web/event-notifications/event-notification-details/SlackNotificationSummary.css:6]: Please use styled-components

File: src/web/index.jsx
Comment (kyleknighted) [inline src/web/index.jsx:3]: please put the spaces back in

**Review by kyleknighted — CHANGES_REQUESTED** [review]
fix up the lint issues and use styled-components then should be good

File: src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx
Comment (waab76) [inline src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx:20]:
```suggestion
      <ReadOnlyFormGroup label="Link Names" value={notification.config.link_names ? `Yes` : `No`} />
```

**Review by waab76 — CHANGES_REQUESTED** [review]

File: src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx
Comment (lingpri) [reply src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx:29]: notify_channel_image

adding the ternary operator removes the icon, with this commit suggestion. please advice, if you would like to remove this behavior.

File: src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx
Comment (lingpri) [reply src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx:5]: I think, there's is no eslint configured with this jenkins build currently.

File: src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx
Comment (waab76) [reply src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx:29]: Did not realize `ReadOnlyFormGroup` had that functionality built in.  Ignore my suggestion for ternary operator.

File: src/web/index.jsx
Comment (Gaya) [inline src/web/index.jsx:11]: We use single quotes according to our style guide.

File: src/web/index.jsx
Comment (Gaya) [inline src/web/index.jsx:17]: Please remove unneeded white space

**Review by Gaya — CHANGES_REQUESTED** [review]
Looks good overall, but there are some things the style guide would have denied. So I think it's good to fix eslint here first.

**Review by Gaya — APPROVED** [review]

**Comment by Gaya** [comment]
Didn't want to wait for re-reviews by Kyle and Rob?

File: src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx
Comment (lingpri) [reply src/web/event-notifications/event-notification-details/SlackNotificationDetails.jsx:35]: @waab76 - I will look into this as a separate task.

**Pull request merged**
